pymeasure/instruments/pendulum/cnt91.py

```diff
@@ -0,0 +1,184 @@
+                    log.info("Buffer was too long, truncated.")
+                break
+        return data
+
+    external_start_arming_source = Instrument.control(
+        "ARM:SOUR?",
+        "ARM:SOUR %s",
+        """
+        Select arming input or switch off the start arming function.
+        Options are 'A', 'B' and 'E' (rear). 'IMM' turns trigger off.
+        """,
+        validator=strict_discrete_set,
+        values={"A": "EXT1", "B": "EXT2", "E": "EXT4", "IMM": "IMM"},
+        map_values=True,
+    )
+
+    external_arming_start_slope = Instrument.control(
+        "ARM:SLOP?",
+        "ARM:SLOP %s",
+        "Set slope for the start arming condition.",
+        validator=strict_discrete_set,
+        values=["POS", "NEG"],
+    )
+
+    continuous = Instrument.control(
+        "INIT:CONT?",
+        "INIT:CONT %s",
+        "Controls whether to perform continuous measurements.",
+        strict_discrete_set,
+        values={True: 1.0, False: 0.0},
+        map_values=True,
+    )
+
+    measurement_time = Instrument.control(
+        ":ACQ:APER?",
+        ":ACQ:APER %f",
+        "Gate time for one measurement in s.",
+        validator=strict_range,
+        values=[2e-9, 1000],  # Programmer's guide 8-92
+    )
+
+    format = Instrument.control(
+        "FORM?",
+        "FORM %s",
+        "Reponse format (ASCII or REAL).",
+        validator=strict_discrete_set,
+        values=["ASCII", "REAL"],
+    )
+
+    interpolator_autocalibrated = Instrument.control(
+        ":CAL:INT:AUTO?",
+        "CAL:INT:AUTO %s",
+        "Controls if interpolators should be calibrated automatically.",
+        strict_discrete_set,
+        values={True: 1.0, False: 0.0},
+        map_values=True,
+    )
+
+    def configure_frequency_array_measurement(self, n_samples, channel):
+        """
+        Configure the counter for an array of measurements.
+
+        :param n_samples: The number of samples
+        :param channel: Measurment channel (A, B, C, E, INTREF)
+        """
+        n_samples = truncated_range(n_samples, [1, self.MAX_BUFFER_SIZE])
+        channel = strict_discrete_set(channel, self.CHANNELS)
+        channel = self.CHANNELS[channel]
+        self.write(f":CONF:ARR:FREQ {n_samples},(@{channel})")
+
+    def buffer_frequency_time_series(
+        self, channel, n_samples, sample_rate, trigger_source=None
+    ):
+        """
+        Record a time series to the buffer and read it out after completion.
+
+        :param channel: Channel that should be used
+        :param n_samples: The number of samples
+        :param sample_rate: Sample rate in Hz
+        :param trigger_source: Optionally specify a trigger source to start the
+            measurement
+        """
+        if self.interpolator_autocalibrated:
+            max_sample_rate = 125e3
+        else:
+            max_sample_rate = 250e3
+        sample_rate = strict_range(sample_rate, [0, max_sample_rate])
```

Comment:
0 as a sample rate is really admissible? I think the lower limit is formed by the upper of `measurement_time`, i.e. 1/1000

Reply:
I agree, added a `MAX_MEASUREMENT_TIME` in 7ec480e to avoid having two dependend magic numbers in the code. Note that `Instrument.control` needs access to this value but it doesn't have access to `self`. Since I didn't find a clean build-in solution to define class attributes, I put this constant in the main body of the module.

Open for ideas of how to improve this.